src/NHibernate/Tool/hbm2ddl/SchemaMetadataUpdater.cs

```diff
@@ -8,17 +9,24 @@ namespace NHibernate.Tool.hbm2ddl
 	// Candidate to be exstensions of ISessionFactory and Configuration
 	public static class SchemaMetadataUpdater
 	{
-		public static void Update(ISessionFactory sessionFactory)
+		public static void Update(ISessionFactoryImplementor sessionFactory)
 		{
-			var factory = (ISessionFactoryImplementor) sessionFactory;
-			var dialect = factory.Dialect;
-			var connectionHelper = new SuppliedConnectionProviderConnectionHelper(factory.ConnectionProvider);
-			factory.Dialect.Keywords.UnionWith(GetReservedWords(dialect, connectionHelper));
+			var dialect = sessionFactory.Dialect;
+			var connectionHelper = new SuppliedConnectionProviderConnectionHelper(sessionFactory.ConnectionProvider);
+			sessionFactory.Dialect.Keywords.UnionWith(GetReservedWords(dialect, connectionHelper));
 		}
 
 		public static void QuoteTableAndColumns(Configuration configuration)
 		{
-			ISet<string> reservedDb = GetReservedWords(configuration.GetDerivedProperties());
+			// Instatiates a new instance of the dialect so doesn't benefit from the Update call.
+			var dialect = Dialect.Dialect.GetDialect(configuration.GetDerivedProperties());
+			QuoteTableAndColumns(configuration, dialect);
```

Comment:
So, the problem was, that the `QuoteTableAndColumns` wasn't even using the keywords loaded into the Dialect.  This was a major problem on the .NET Standard branch because there is no functioning auto-importing, so schema names weren't getting quoted at all, because the DbConnection based GetSchema had to return an empty list.

The auto-import is enabled if auto-quoting is enabled, and auto-import takes place first during the session factory construction.

I think the only paths that use auto-quoting without going through the session factory construction, and thus the auto-importing, are some unit tests that are trying to test just that feature.  For those the auto-quoting to conflate auto-importing, it's not allowing the tests the isolation they may desire.